Source/Scene/Multiple3DTileContent.js

```diff
@@ -0,0 +1,540 @@
+//import defaultValue from "../Core/defaultValue.js";
+import defined from "../Core/defined.js";
+import destroyObject from "../Core/destroyObject.js";
+import Request from "../Core/Request.js";
+import RequestScheduler from "../Core/RequestScheduler.js";
+import RequestState from "../Core/RequestState.js";
+import RequestType from "../Core/RequestType.js";
+import RuntimeError from "../Core/RuntimeError.js";
+import when from "../ThirdParty/when.js";
+import Cesium3DTileContentType from "./Cesium3DTileContentType.js";
+import Cesium3DTileContentFactory from "./Cesium3DTileContentFactory.js";
+import preprocess3DTileContent from "./preprocess3DTileContent.js";
+
+/**
+ * A collection of contents for tiles that use the <code>3DTILES_multiple_contents</code> extension.
+ * <p>
+ * Implements the {@link Cesium3DTileContent} interface.
+ * </p>
+ *
+ * @see {@link https://github.com/CesiumGS/3d-tiles/tree/3d-tiles-next/extensions/3DTILES_multiple_contents/0.0.0|3DTILES_multiple_contents extension}
+ *
+ * @alias Multiple3DTileContent
+ * @constructor
+ *
+ * @param {Cesium3DTileset} tileset The tileset this content belongs to
+ * @param {Cesium3DTile} tile The content this content belongs to
+ * @param {Resource} tilesetResource The resource that points to the tileset. This will be used to derive each inner content's resource.
+ * @param {Object} extensionJson The <code>3DTILES_multiple_contents</code> extension JSON
+ *
+ * @private
+ */
+export default function Multiple3DTileContent(
+  tileset,
+  tile,
+  tilesetResource,
+  extensionJson
+) {
+  this._tileset = tileset;
+  this._tile = tile;
+  this._tilesetResource = tilesetResource;
+  this._contents = [];
+
+  var contentHeaders = extensionJson.content;
+  this._innerContentHeaders = contentHeaders;
+  this._requestsInFlight = 0;
+
+  /**
+   * This is set to true if one of the inner contents are canceled.
+   * @type {Boolean}
+   * @private
+   */
+  this.canceled = false;
+
+  var contentCount = this._innerContentHeaders.length;
+  this._arrayFetchPromises = new Array(contentCount);
+
+  this._innerContentResources = new Array(contentCount);
+  this._serverKeys = new Array(contentCount);
+
+  var priorityFunction = function () {
+    return tile.priority;
+  };
+  for (var i = 0; i < contentCount; i++) {
+    var contentResource = tilesetResource.getDerivedResource({
+      url: contentHeaders[i].uri,
+      preserveQueryParameters: true,
+    });
+
+    var serverKey = RequestScheduler.getServerKey(
+      contentResource.getUrlComponent()
+    );
+
+    var request = new Request({
+      throttle: true,
+      throttleByServer: true,
+      type: RequestType.TILES3D,
+      priorityFunction: priorityFunction,
```

Comment:
You've probably thought about this a lot already, but is there a clean way for `Cesium3DTile` and `Multiple3DTileContent` to use a common function for content requests?

Reply:
Now that I moved the request creation around in `Multiple3DTileContent.js`, I don't think the two approaches are similar enough for a refactor like this to be effective. I'm going to hold off for now.